### PR TITLE
fix: backport reused connection fix

### PR DESCRIPTION
**Pull request by salzhrani**
this PR is a back-port of the work done in [this](https://github.com/fastify/fastify/pull/4133) PR to the 3.x branch
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: fastify.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-const VERSION = '3.29.0'
+const VERSION = '3.29.1'
```
Comment (jsumners) [inline fastify.js:3]: Can you explain this change? If I'm not mistaken, we have some sort of build process that will automatically bump this (believe me, I don't like it).

Comment (mcollina) [reply fastify.js:3]: I don't think I actually bumped it last time, so the tests are failing now.

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Comment by mcollina** [comment]
```
​ FAIL ​ test/request-error.test.js 1 failed of 31 498.848ms
 ✖ should match pattern provided
```

CI is failing on node 10

**Comment by salzhrani** [comment]
it seems in node v10 calling destroy [here](https://github.com/fastify/fastify/pull/4217/files#diff-a36b53a0e851347821d855f09722921b64d2792750d1deaab126fe54964498a9R590) can cause the stream not to drain, and no data chunks reach the client. hence the failed test. I would try `end` instead of `destroy` but it causes another test to fail. not sure where to go from here

**Comment by mcollina** [comment]
Maybe make the switch conditional depending on the major version of Node.js.

**Comment by salzhrani** [comment]
Can you elaborate please?

**Comment by mcollina** [comment]
Let's try skipping the test if it's node v10.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by salzhrani** [comment]
suggestion applied @mcollina

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**